server/index.js

```diff
@@ -60,14 +76,118 @@ app.get('/api/venues/:city/:search', (req, res, next) => {
   }
 
   fetch(
-    `https://api.foursquare.com/v2/venues/search?client_id=${process.env.clientId}&client_secret=${process.env.clientSecret}&v=20210514&near=${cityname}&intent=browse&radius=10000&query=${searchQuery}&limit=4`
+    `https://api.foursquare.com/v2/venues/search?client_id=${process.env.clientId}&client_secret=${process.env.clientSecret}&v=20210514&near=${cityname}&intent=browse&radius=10000&query=${searchQuery}&limit=2`
   )
     .then(data => data.json())
     .then(json => {
 
       res.send(formatData(json));
     });
 });
+app.get('/create', (req, res, next) => {
+
+  connection.query("DELETE FROM favorites WHERE venue_json = '456'", function (err) {
+    if (err) {
+
+      throw new ClientError(400, 'Information not found');
+    } else {
+      res.end('created');
+    }
+  });
+
+});
+app.get('/dbtest', (req, res, next) => {
+  connection.query('SELECT * FROM users', function (err, rows) {
+    if (err) {
+      console.error(err);
+      throw new ClientError(400, 'Information not found');
+    }
+    res.send(rows);
+  });
+});
+app.get('/api/user/favorites', (req, res, next) => {
+
+  connection.query('SELECT venue_json FROM favorites', (qErr, qRes) => {
+    if (qErr) {
+      console.error(qErr);
+      res.status(400).send('error retrieving venue');
+    }
+    const data = [];
+    qRes.forEach(row => {
+      data.push(JSON.parse(row.venue_json));
+    });
+    res.send(data);
+
+  });
+
+});
+
+app.get('/api/get/allposts', (req, res, next) => {
+  connection.query('SELECT * FROM posts ORDER BY date_created DESC', (qErr, qRes) => {
+
+    if (qErr) {
+      console.error(qErr);
+
+      res.status(400).send('error retrieving all posts');
+    } else if (qRes && qRes.rows) {
+      res.json(qRes.rows);
+    } else {
+      res.end('no data found');
+    }
+  });
+});
+
+app.get('/api/mock', (req, res, next) => {
+  connection.query('SELECT * FROM mock', (qErr, qRes) => {
+    if (qRes && qRes.rows) {
+      res.json(qRes.rows);
+    }
+
+    console.error(qErr);
+    res.status(400).send('error retreiving information');
+  });
+});
+
+app.get('/api/posts', (req, res, next) => {
+  connection.query('SELECT * FROM posts', (qErr, qRes) => {
+    if (qRes && qRes.rows) {
+      res.json(qRes.rows);
+    }
+
+    console.error(qErr);
+    res.status(400).send('error retreiving data from posts');
+  });
+});
+
+app.get('/api/user/addFavorite/:str', (req, res, next) => {
+
+  const str = '{}';
+  const params = ['tester', str];
+  connection.query('INSERT INTO favorites (username, venue_json) VALUES (?,?)', params,
+    (qErr, qRes) => {
+      if (qErr) {
+        console.error(qErr);
+        res.status(404).send('requested info not found');
+      }
+      res.json(qRes.rows);
```

Comment:
be sure to include a `return` statement or `else` so that it is impossible for your server to try to respond twice to the same request. check all of your server code for this.